### PR TITLE
Sqs terraform

### PR DESCRIPTION
**Pull request by MattPinelli**

### CODE REVIEW

File: terraform/iam.tf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 locals {
   ec2_volume_arn = "arn:aws:ec2:${var.aws_region}:${data.aws_caller_identity.current.account_id}:volume/*"
-  snapshot_arn = "arn:aws:ec2:${var.aws_region}::snapshot/*"
+  snapshot_arn   = "arn:aws:ec2:${var.aws_region}::snapshot/*"
```
Comment (Stiffni) [inline terraform/iam.tf]: is this supposed to be there?

Comment (MattPinelli) [reply terraform/iam.tf]: What do you mean, the snapshot_arn variable?

File: terraform/iam.tf [hunk 2]
```diff
@@ -111,7 +137,52 @@ resource "aws_iam_role_policy" "attach_bakery_snapshot" {
...
+      type        = "Service"
+      identifiers = ["sqs.amazonaws.com", "s3.amazonaws.com"]
+    }
+
+    actions = [
+      "kms:Decrypt",
+      "kms:GenerateDataKey",
+    ]
+
+    resources = ["*"]
```
Comment (jtgrohn) [inline terraform/iam.tf]: can this be locked down further?

File: terraform/iam.tf [hunk 3]
```diff
@@ -111,7 +137,52 @@ resource "aws_iam_role_policy" "attach_bakery_snapshot" {
...
+    principals = {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"]
+    }
+
+    actions = [
+      "kms:*",
+    ]
+
+    resources = ["*"]
```
Comment (jtgrohn) [inline terraform/iam.tf]: this too?

File: terraform/iam.tf [hunk 4]
```diff
@@ -111,7 +137,52 @@ resource "aws_iam_role_policy" "attach_bakery_snapshot" {
...
+  statement {
+    effect = "Allow"
+
+    principals = {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"]
+    }
+
+    actions = [
+      "kms:*",
```
Comment (jtgrohn) [inline terraform/iam.tf]: does it really need *?

File: terraform/iam.tf (hunk 2 above)
Comment (MattPinelli) [reply terraform/iam.tf]: I'll try - I was afraid of a circular dependency.

File: terraform/iam.tf (hunk 3 above)
Comment (MattPinelli) [reply terraform/iam.tf]: No, doesn't look like it. I get a circular dependency error.

File: terraform/iam.tf (hunk 4 above)
Comment (MattPinelli) [reply terraform/iam.tf]: I think so (this is the permission that allows administration of the key). I'll double check.

File: terraform/iam.tf (hunk 1 above)
Comment (Stiffni) [reply terraform/iam.tf]: Oh no, the extra spacing

Comment (MattPinelli) [reply terraform/iam.tf]: Oh yeah, I have a terraform formatter. It likes everything to be in line.

**Pull request merged**